src/ex2/pokemonClient.js

```diff
@@ -0,0 +1,147 @@
+class PokemonClient {
+  constructor() {
+    this.alreadyFetchedId = null;
+    this.pokemonsId = {};
+    this.listOfFavoritePokemon = [
+      "bulbasaur",
+      "charizard",
+      "onix",
+      "nidoking",
+      "vileplume",
+    ];
+    this.id = "pokemon fetch";
+    this.api = "https://pokeapi.co/api/v2/pokemon/";
+  }
+
+  handleFailedRequest(allRequestsResults) {
+    let failedIds;
+    let failedRquest;
+
+    failedRquest = allRequestsResults.filter((result) => {
+      return result.includes("not found");
+    });
+
+    if (failedRquest.length > 1) {
+      failedIds = failedRquest.map(removeAllExceptNumbers);
+      failedIds = failedIds.join(",");
+
+      failedRquest = ["Failed to fetch pokemon with this input: " + failedIds];
+    }
+
+    return failedRquest;
+  }
+  createPokemonTask(name, types) {
+    return (
+      "Catch " + name + " the " + this.getPokemonType(types) + " type pokemon"
+    );
+  }
+  async fetchPokemon(id) {
+    const api = this.api + id;
+    let text;
+    try {
+      const rawResponse = await fetch(api);
+      const content = await rawResponse.json();
+      this.pokemonsId[content.name] = id;
+
+      text = this.createPokemonTask(content.name, content.types);
+    } catch (e) {
```

Comment:
Should handle response & errors using http status code.
this error handling `catch` is only happening because of your action of `rawResponse.json();` which fails the code due to no json in the response.

So it seems like it is working but in fact it's not handled properly/safely